### PR TITLE
drivers: sensor: bmi270: add pm device support

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch introduces support for suspending and resuming the device using power management APIs.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]
Please implement runtime PM, CONFIG_PM_DEVICE is useless.

**Comment by maxd-nordic** [comment]
@gmarull I reworked this. Could you have another look?

**Comment by maxd-nordic** [comment]
@JordanYates

File: drivers/sensor/bmi270/bmi270_trigger.c [hunk 1]
```diff
@@ -84,6 +84,24 @@ static void bmi270_thread_cb(const struct device *dev)
 	}
 }
 
+/* claim the device for PM_DEVICE_RUNTIME if triggers are configured */
+static int handle_pm_device_claim(const struct device *dev)
```
Comment (JordanYates) [inline drivers/sensor/bmi270/bmi270_trigger.c:88]: What is the purpose of this function? Why not call the PM functions directly?

**Review by JordanYates — COMMENTED** [review]
How does this work for `bmi270_attr_set`? I don't see any PM related code there.

File: drivers/sensor/bmi270/bmi270_trigger.c (hunk 1 above)
Comment (maxd-nordic) [reply drivers/sensor/bmi270/bmi270_trigger.c]: I just moved to stuff into the calling function

**Pull request closed without merging**